### ISSUE TITLE
Consider minimise shaded jar to reduce size of final artifact

### ISSUE THREAD

**Issue by brenuart** [logstash/logstash-logback-encoder#119]
The build makes use of the `maven-shade-plugin` to embed the following dependencies in the final artifact:
- `commons-lang:common-lang:2.6`
- `commons-io:commons-io:2.4`
- `com.lmax:disruptor:3.3.2`

The code base makes use of only a couple of utility methods from both `commons-lang` and `commons-io` - but the entire libraries are included making the final artifact bigger than necessary.

The `maven-shade-plugin` could be configured to minimise the final jar by including only the classes that are actually required. This could reduce the size from about *750Kb* down to *450Kb*.

```
                <plugin>
                    <groupId>org.apache.maven.plugins</groupId>
                    <artifactId>maven-shade-plugin</artifactId>
                    <version>2.2</version>
                    <configuration>
                        <shadedArtifactAttached>false</shadedArtifactAttached>
                        <createDependencyReducedPom>true</createDependencyReducedPom>
                        <minimizeJar>false</minimizeJar>
```

Don't you think it is worth to activate this option?

**Comment by brenuart** [comment]
Concerning `commons-io`, only very simple utility methods like `closeQuitely()` are used. They could easily be implemented inside the projet and the dependency removed.

Other methods are used by test cases - if not reimplemented, `commons-io` could simply be downgraded to a test dependency.

**Comment by philsttr** [comment]
Yes! Definitely. Totally agree with everything you mentioned.

**Comment by brenuart** [comment]
The easiest is letting the `maven-shade-plugin`do the work for us...
Do you need a PR for that?

**Comment by philsttr** [comment]
I can handle this

**Comment by philsttr** [comment]
Released in 4.5
